### PR TITLE
Explicitly rename junos_ospf to junos_ospfv2

### PR DESCRIPTION
**Pull request by danielmellado**
This PR renames ospf to ospfv2 in order to be explicitly clear about
the protocol used.

### CODE REVIEW

File: plugins/modules/junos_ospfv2.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 
 DOCUMENTATION = """
-module: junos_ospf
+module: junos_ospfv2
 short_description: OSPF resource module
```
Comment (rohitthakur2590) [inline plugins/modules/junos_ospfv2.py:36]: short_description: OSPFV2 resource module

File: docs/junipernetworks.junos.junos_ospfv2.rst [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-.. _junipernetworks.junos.junos_ospf_:
+.. _junipernetworks.junos.junos_ospfv2_:
 
 
 *****
-junipernetworks.junos.junos_ospf
+junipernetworks.junos.junos_ospfv2
 *****
 
 **OSPF resource module**
```
Comment (rohitthakur2590) [inline docs/junipernetworks.junos.junos_ospfv2.rst:9]: OSPFV2

File: docs/junipernetworks.junos.junos_ospfv2.rst [hunk 3]
```diff
@@ -821,7 +821,7 @@ Examples
     # admin# show protocols ospf
 
     - name: Merge Junos OSPF config
```
Comment (rohitthakur2590) [inline docs/junipernetworks.junos.junos_ospfv2.rst:823]: - name: Merge Junos OSPFV2 config

File: plugins/modules/junos_ospfv2.py [hunk 4]
```diff
@@ -218,7 +218,7 @@
 # admin# show protocols ospf
 
 - name: Merge Junos OSPF config
```
Comment (rohitthakur2590) [inline plugins/modules/junos_ospfv2.py:220]: OSPFV2

**Comment by danielmellado** [comment]
recheck

**Comment by danielmellado** [comment]
recheck

**Pull request merged**